app/controllers/vm_cloud_controller.rb

```diff
@@ -91,6 +91,91 @@ def resize_field_changed
     end
   end
 
+  def live_migrate
+    assert_privileges("instance_live_migrate")
+    @record = find_by_id_filtered(VmOrTemplate, params[:id]) # Set the VM object
+    drop_breadcrumb(
+      :name => _("Live Migrate Instance '%{name}'") % {:name => @record.name},
+      :url  => "/vm_cloud/live_migrate"
+    ) unless @explorer
+
+    @edit = {}
+    @edit[:key] = "vm_migrate__#{@record.id}"
+    @edit[:vm_id] = @record.id
+    @edit[:explorer] = true if params[:action] == "x_button" || session.fetch_path(:edit, :explorer)
+    session[:edit] = @edit
+    @in_a_form = true
+    @refresh_partial = "vm_common/live_migrate"
+  end
+  alias_method :instance_live_migrate, :live_migrate
+
+  def live_migrate_form_fields
+    assert_privileges("instance_live_migrate")
+    @record = find_by_id_filtered(VmOrTemplate, params[:id])
+    clusters = []
+    hosts = []
+    @record.ext_management_system.ems_clusters.each { |c| clusters << {:id => c.id, :name => c.name} }
```

Comment:
@mansam In this line and the next, I think the drop downs should be populated only with clusters and hosts that the user is allowed to see based on his RBAC.  @AparnaKarve can you point to an example?